irohad/subscription/common.hpp

```diff
@@ -10,7 +10,7 @@
 #include <mutex>
 #include <shared_mutex>
 
-#if __clang__
+#ifdef __APPLE__
```

Comment:
```suggestion
#if defined(__APPLE__) and defined(__clang__)
```
GCC also works on Apple

Reply:
```reinterpret_pointer_cast``` is not intrinsic. It locate in ```libc++```. So it doesn't depend on compiler, just library implementation in the platform.